src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -686,38 +951,43 @@ function AnalyticsService:LogResourceEvent(data: ResourceEvent): { [any]: any }
 		elseif typeof(data.amount) ~= "number" then
 			return reject("amount is required")
 		end
-		
-		GameAnalytics:addResourceEvent(data.userId, {
+
+		if not table.find(self._resourceEventTypes, data.eventType) then
+			self._resourceEventTypes[#self._resourceEventTypes + 1] = data.eventType
+			GameAnalytics.Events:setAvailableResourceItemTypes(self._resourceEventTypes) -- Update the SDK
+		end
+
+		self:_wrapper("addResourceEvent", data.userId, {
 			-- FlowType is Sink by default
 			flowType = data.flowType,
 			currency = data.currency,
 			amount = data.amount,
 			itemType = data.eventType,
-			itemId = data.itemId
+			itemId = data.itemId,
 		})
-		
-		return resolve()
+
+		resolve()
```

Comment:
```suggestion
		return resolve()
```
The linter/type checker complains if there's a return type defined for a function, but the promise is resolved/rejected without returning the function (For some reason)? It's an invalid error since we know we're returning a promise, but the code checker flags it for some reason.